x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/analyze/BlobAnalyzeAction.java

```diff
@@ -230,6 +258,7 @@ private static class BlobAnalysis {
                 checksumStart = randomLongBetween(0L, request.targetLength);
                 checksumEnd = randomLongBetween(checksumStart + 1, request.targetLength + 1);
             }
+            doEarlyCopy = random.nextBoolean();
```

Comment:
Nice, but consider this an optional bonus feature, I'd be happy even if we only covered the copy-after-write path. Note that we can only track the copy as an expected blob in `RepositoryAnalyzeAction` if it doesn't potentially fail because of an early copy, so we'd need to pull this parameter up to the top level too.

Reply:
Would you prefer that I remove early copy from this PR then?